From e0cb054fe0b19f78babff33b5ff082ff2c70985f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 05 Jul 2021 12:55:52 +0300
Subject: [PATCH] [SW-46517] preboot: greco: remove sram scramble

In secureboot flow, preboot will be moved from scratchpad SRAM to
SRAM. In order to achieve this, SRAM scrambling init which
is done in preboot is moved to ppboot as it needs to be done before
SRAM is used by FW. This patch removes SRAM scrambler enabling function
from preboot.

Change-Id: I00eeaea2754c2dac8e07f2155e359afaad1cdbf9
---

diff --git a/arch/greco_arc_gnu/asic.c b/arch/greco_arc_gnu/asic.c
index 8ac4a73..aee6d37 100644
--- a/arch/greco_arc_gnu/asic.c
+++ b/arch/greco_arc_gnu/asic.c
@@ -7,14 +7,6 @@
 
 #include <psoc_global_conf_regs.h>
 #include <psoc_gpio0_regs.h>
-#include <dcon0_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon0_hbw_rtr_if1_rtr_h3_regs.h>
-#include <dcon1_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon1_hbw_rtr_if1_rtr_h3_regs.h>
-#include <dcon2_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon2_hbw_rtr_if1_rtr_h3_regs.h>
-#include <dcon3_hbw_rtr_if0_rtr_h3_regs.h>
-#include <dcon3_hbw_rtr_if1_rtr_h3_regs.h>
 
 #include <bitops.h>
 #include <preboot.h>
@@ -56,24 +48,6 @@
 	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
 	val &= ~BIT(2);
 	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
-}
-
-/* initialize sram scrambling before u-boot is downloaded */
-static void sram_scrambling_init(void)
-{
-	regmap_writel(1, mmDCON0_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON0_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON1_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON1_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON2_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON2_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON3_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON3_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-}
-
-void hl_ram_init(void)
-{
-	sram_scrambling_init();
 }
 
 u32 hl_spi_baudr_get(void)
